src/workbench/InfoDialog.tsx

```diff
@@ -4,33 +4,63 @@
  * SPDX-License-Identifier: MIT
  */
 import { Button } from "@chakra-ui/button";
+import { HStack, Link, Stack, Text, VStack } from "@chakra-ui/layout";
 import {
-  Flex,
-  Link,
-  List,
-  ListItem,
-  Stack,
-  Text,
-  VStack,
-} from "@chakra-ui/layout";
-import { Modal, ModalBody, ModalContent, ModalOverlay } from "@chakra-ui/modal";
-import { RiExternalLinkLine, RiFeedbackLine } from "react-icons/ri";
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalOverlay,
+} from "@chakra-ui/modal";
+import { Icon } from "@chakra-ui/react";
+import { ReactNode, useEffect, useState } from "react";
+import { RiExternalLinkLine } from "react-icons/ri";
+import { FormattedMessage } from "react-intl";
 import ModalCloseButton from "../common/ModalCloseButton";
+import { fetchContent } from "../common/sanity";
+import YoutubeVideoEmbed, { YoutubeVideo } from "../common/YouTubeVideo";
 import { useDeployment } from "../deployment";
+import { useLogging } from "../logging/logging-hooks";
+import { useSettings } from "../settings/settings";
 
 interface InfoDialogProps {
   isOpen: boolean;
   info?: boolean;
   onClose: () => void;
-  switchToInfoDialog: () => void;
 }
 
-const InfoDialog = ({
-  isOpen,
-  onClose,
-  switchToInfoDialog,
-}: InfoDialogProps) => {
+const InfoDialog = ({ isOpen, onClose }: InfoDialogProps) => {
   const { guideLink } = useDeployment();
+  const [welcomeVideo, setWelcomeVideo] = useState<YoutubeVideo | undefined>();
+  const [loadError, setLoadError] = useState<boolean>(false);
+  const [{ languageId }] = useSettings();
+  const logging = useLogging();
+  useEffect(() => {
+    const adaptContent = (result: any): YoutubeVideo | undefined => {
+      if (result.length === 1 && result[0].welcomeVideo) {
+        return result[0].welcomeVideo;
+      }
+    };
+    const query = (languageId: string): string => {
+      if (!languageId.match(/^[a-z-]+$/g)) {
+        throw new Error("Invalid language id.");
+      }
+      // Ready to use languageId here if the alt field is made translatable.
+      return `
+        *[_id == "pythonEditorConfig"]{
```

Comment:
We should exclude drafts here (see other query).

Reply:
[`56d8b38`](https://github.com/microbit-foundation/python-editor-next/pull/794/commits/56d8b38005280e77d983778e19a0083697db9312).